source/adapters/level_zero/command_buffer.cpp

```diff
@@ -241,9 +299,25 @@ ur_result_t enqueueCommandBufferMemCopyHelper(
       CommandType, CommandBuffer, NumSyncPointsInWaitList, SyncPointWaitList,
       false, RetSyncPoint, ZeEventList, ZeLaunchEvent));
 
+  // The chooseCommandList() will modify MComputeCommandListEmpty and
+  // MCopyCommandListEmpty so isFirstNode needs to be called beforehand.
+  bool IsFirstNode{CommandBuffer->isFirstNode()};
+
   ze_command_list_handle_t ZeCommandList =
       CommandBuffer->chooseCommandList(PreferCopyEngine);
 
+  // Create a sync point if both Copy and Compute in-order command lists are
+  // used.
+  if (CommandBuffer->IsInOrderCmdList &&
```

Comment:
This bit of code is copy pasted a lot. It can probably be refactored into its own function.

Reply:
Yep, I've put some of this logic inside `createSyncPointBetweenCopyAndCompute()` and in `appendWaitBetweenCopyAndCompute()`. Let me know what you think.